Imports availableClassrooms into frontend

oscar-luna opened the pull request:
closes #18 

Moves data into a variable called availableClassrooms using useEffect


https://courses-oscar-luna.dokku-01.cs.ucsb.edu

Temporarily displaying the classrooms in the frontend using this style
<img width="920" alt="Screenshot 2025-05-27 at 7 24 22 AM" src="https://github.com/user-attachments/assets/633d31e3-8210-4226-8464-da8a4d75b0e2" />

oscar-luna committed:
initial api query with hook

oscar-luna committed:
eslint fix

oscar-luna committed:
adds tests

oscar-luna committed:
format

oscar-luna committed:
redo test method

oscar-luna committed:
more test edits

oscar-luna committed:
more test edits

oscar-luna committed:
more test edits

oscar-luna committed:
stryker fix

oscar-luna committed:
edits

oscar-luna committed:
add line

oscar-luna committed:
tab

oscar-luna committed:
add empty line

oscar-luna committed:
comments and small edits

oscar-luna committed:
removed unnecessary line

alecsekimoto1 reviewed (approved):
LGTM

alecsekimoto1 reviewed (dismissed):
LGTM

ameymwalimbe reviewed (commented):
Let me know if there is another issue that handles this, but I don't think the current implementation of CourseOverTimeBuildingsSearchForm.js accepts any availableClassrooms prop. Ideally, I'd like to see the availableClassrooms integrated into the Search Form, as well as a storybook link so the staff can easily verify that these changes work.

pconrad commented:
I'm looking at this PR and it's linked issue [18](https://github.com/ucsb-cs156-s25/proj-courses-s25-01/issues/18) and I'm having trouble seeing the big picture.

To be fair, the original issue was part of a larger epic, and it isn't explained very well in the issue itself.  I concede this point.  

But: in terms of implementing it, were you able to figure out what it's purpose is the larger scheme of things?

More to the point: what should I be looking for when testing this issue?

I see no dokku deployment, and I see no storybook entry.   I'm wondering if this should have been a standalone PR at all?

@ameymwalimbe asked similar questions in his code review a couple of days ago:

> Let me know if there is another issue that handles this, but I don't think the current implementation of 
> CourseOverTimeBuildingsSearchForm.js accepts any availableClassrooms prop. Ideally, I'd like to see the 
> availableClassrooms integrated into the Search Form, as well as a storybook link so the staff can easily verify that 
> these changes work.

And there's been no follow up or response.  Do you plan to move forward with this PR or abandon it?

I'm marking PR26 as stalled because it doesn't appear that any progress is being made on it.   
Does a later PR depend on this one?  If so, making that clear in the PR description would help a lot.

oscar-luna commented:
This PR felt like a step into the EPIC of courese. The task and complexity here came from integrating the classroom api call to bring in the variable for the dropdown.

 I assumed in the final PR for the epic #9 we were going to combine the availableClassrooms variable with the  Dropdown plugin in #30 
The final PR would have been bringing availableClassrooms into CourseOverTimeBuildingsSearchForm.js and figuring a way to filter the course data to coincide with just the room. I also heard they may have been an API method.

In my testing I did display the bare data of availableCLassrooms and it does contain whats expected.

I believe I did accomplish the goals of the Issue request and I hope we can get points for it. 

 The final PR for #9 would be addressing the issues of it's practicality and making a storybook for it. Otherwise it's just an odd nonfunctional dropdown menu or text field.

oscar-luna committed:
Merge branch 'main' into classrooms_frontend

oscar-luna commented:
Made the above fix temporarily displaying the classrooms in the frontend 

<img width="1110" alt="Screenshot 2025-05-27 at 5 30 45 PM" src="https://github.com/user-attachments/assets/bf1e6ccb-278b-4517-bdc5-e0f4e924f773" />

ameymwalimbe committed:
Merge branch 'main' into classrooms_frontend

ameymwalimbe reviewed (approved):
LGTM
